sdk/core/Azure.Core/tests/HttpPipelineBuilderTest.cs

```diff
@@ -137,7 +137,7 @@ public async Task UsesAssemblyNameAndInformationalVersionForTelemetryPolicySetti
             HttpPipeline pipeline = HttpPipelineBuilder.Build(options);
 
             var message = pipeline.CreateMessage();
-            var userAgent = new TelemetryDetails(typeof(string).Assembly);
+            var userAgent = new TelemetryDetails(typeof(string).Assembly, default);
```

Comment:
Is it required to pass default now? If so, why?

Reply:
Only for internal code - because of the ambiguity of the optional arguments. Client or customer code will not need to do this.

Reply:
Is it worth changing the constructor definition to avoid this?  There are FDG guidelines that recommend a pattern here, e.g. in section 5.1, "AVOID having two overloads of the same method that both use default parameters" and "DO move all default parameters to the new, longer overload when adding optional parameters to an existing method."  The latter is complicated by the longer overload being internal, but if you have to pass a value anyway if you have access to internal code, why provide a default value for the parameter?

Reply:
I don't think the guideline applies here because the new parameter should not be exposed in the existing ctor and the existing method should call the internal ctor with the default value to avoid duplicating the code.